cpg_workflows/large_cohort/relatedness.py

```diff
@@ -125,17 +132,25 @@ def flag_related(
 def _compute_sample_rankings(ht: hl.Table) -> hl.Table:
     """
     Orders samples by hard filters and coverage and adds rank, which is the lower,
-    the better.
+    the better. The ranking is based on either 'var_data_chr20_mean_dp' or
+    'autosomal_mean_dp' depending on the 'inf_ploidy_using_var' configuration.
 
-    @param ht: table with a `var_data_chr20_mean_dp` and `filters` fields.
+    @param ht: table with either `var_data_chr20_mean_dp` or `autosomal_mean_dp` and `filters` fields.
     @return: table ordered by rank, with the following row fields:
         `rank`, `filtered`
     """
+    use_only_variants = get_config()['large_cohort']['pca_background'].get(
```

Comment:
`_compute_sample_rankings` may be called from other contexts, now or in the future.

I would rather determine which column name to use by introspecting `ht`. Either by checking the global annotations via something like `hl.eval(ht[variants_only_x_ploidy])` (and checking `variants_only_y_ploidy` too) or perhaps by checking whether a `var_data_chr20_mean_dp` column exists. The former is probably better.

Reply:
I've changed it to the following in the latest commit: 
```python
if 'var_data_chr20_mean_dp' in ht.row:
        ranking_column = 'var_data_chr20_mean_dp'
    else:
        ranking_column = 'autosomal_mean_dp'
```

If also occurs in `add_soft_filters` inside `sample_qc.py`:
```python
if 'var_data_chr20_mean_dp' in ht.row:
        autosomal_coverage_colname = 'var_data_chr20_mean_dp'
    else:
        autosomal_coverage_colname = 'autosomal_mean_dp'
    ht = add_filter(
        ht,
        ht[autosomal_coverage_colname] < cutoffs['min_coverage'],
        'low_coverage',
    )
```